lib/utils/qemu-ts.ts

```diff
@@ -42,20 +46,39 @@ export async function installQemuIfNeeded(
 
 /**
  * Check whether the Docker daemon (including balenaEngine) requires explicit
- * QEMU emulation setup. Note that Docker for Mac (macOS), and reportedly also
- * Docker for Windows, have built-in support for binfmt_misc, so do not require
- * explicity QEMU setup. References:
+ * QEMU emulation setup. Note that Docker Desktop (Docker for Mac), and
+ * reportedly also Docker for Windows, have built-in support for binfmt_misc,
+ * so they do not require explicity QEMU setup. References:
  * - https://en.wikipedia.org/wiki/Binfmt_misc
  * - https://docs.docker.com/docker-for-mac/multi-arch/
  * - https://www.ecliptik.com/Cross-Building-and-Running-Multi-Arch-Docker-Images/
  * - https://stackoverflow.com/questions/55388725/run-linux-arm-container-via-qemu-binfmt-misc-on-docker-lcow
  *
  * @param docker Dockerode instance
  */
-async function platformNeedsQemu(docker: Dockerode): Promise<boolean> {
+async function platformNeedsQemu(
+	docker: Dockerode,
+	logger: Logger,
+): Promise<boolean> {
 	const dockerInfo = await docker.info();
-	// Docker for Mac reports dockerInfo.OSType as 'linux'
+	// Docker Desktop with Docker Engine 19.03 reports:
+	//     OperatingSystem: Docker Desktop
+	//     OSType: linux
+	// Docker for Mac with Docker Engine 18.06 reports:
+	//     OperatingSystem: Docker for Mac
+	//     OSType: linux
+	// On Ubuntu (standard Docker installation):
+	//     OperatingSystem: Ubuntu 18.04.2 LTS (containerized)
+	//     OSType: linux
 	// https://stackoverflow.com/questions/38223965/how-can-i-detect-if-docker-for-mac-is-installed
-	const isDockerForMac = /Docker for Mac/i.test(dockerInfo.OperatingSystem);
-	return dockerInfo.OSType === 'linux' && !isDockerForMac;
+	const isDockerDesktop = /(?:Docker Desktop)|(?:Docker for Mac)/i.test(
+		dockerInfo.OperatingSystem,
+	);
+	if (isDockerDesktop) {
+		logger.logInfo(stripIndent`
+			Docker Desktop detected (daemon architecture: "${dockerInfo.Architecture}")
+			Docker itself will determine and enable architecture emulation if required,
+			without balena-cli intervention and regardless of the --emulated option.`);
+	}
+	return !isDockerDesktop;
```

Comment:
Sample build output with these changes:
```
$ ./bin/balena build -a test-rpi simple_container --logs --nocache --emulated
[Info]    Creating default composition with source: simple_container
[Info]    Building for armv7hf/raspberrypi3
[Build]   Building services...
[Build]   main Preparing...
[Info]    Docker Desktop detected (daemon architecture: "x86_64")
[Info]    Docker itself will determine and enable architecture emulation if required,
[Info]    without balena-cli intervention and regardless of the --emulated option.
[Build]   main Step 1/3 : FROM arm32v7/busybox
[Build]   main  ---> 723ce4685637
[Build]   main Step 2/3 : RUN uname -a
[Build]   main  ---> Running in f54620a94693
[Build]   main Linux f54620a94693 4.9.184-linuxkit #1 SMP Tue Jul 2 22:58:16 UTC 2019 armv7l GNU/Linux
```

Reply:
Generally what I tend to with these multiline message is to add a tab or a couple of spaces for every line that isn't the first, this seems to make things more clear that it's part of the same message.

Reply:
Good idea. I've added a couple of spaces:
```
[Info]    Docker Desktop detected (daemon architecture: "x86_64")
[Info]      Docker itself will determine and enable architecture emulation if required,
[Info]      without balena-cli intervention and regardless of the --emulated option.
```

Reply:
Looks good, great work.